cmd/ui/src/views/DatabaseManagement/DatabaseManagement.tsx

```diff
@@ -0,0 +1,307 @@
+    FormControlLabel,
+    FormGroup,
+    FormHelperText,
+    Typography,
+    useTheme,
+} from '@mui/material';
+import { ContentPage, apiClient } from 'bh-shared-ui';
+import { useReducer } from 'react';
+import ConfirmationDialog from './ConfirmationDialog';
+import { useMutation } from 'react-query';
+import { useSelector } from 'react-redux';
+import { selectTierZeroAssetGroupId } from 'src/ducks/assetgroups/reducer';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCircleXmark, faCircleExclamation } from '@fortawesome/free-solid-svg-icons';
+
+type DataTypes = {
+    collectedGraphData: boolean;
+    highValueSelectors: boolean;
+    fileIngestHistory: boolean;
+    dataQualityHistory: boolean;
+};
+
+const initialState: State = {
+    collectedGraphData: false,
+    highValueSelectors: false,
+    fileIngestHistory: false,
+    dataQualityHistory: false,
+
+    noSelectionError: false,
+    mutationError: false,
+    showSuccessMessage: false,
+
+    openDialog: false,
+};
+
+type State = {
+    // checkbox state
+    collectedGraphData: boolean;
+    highValueSelectors: boolean;
+    fileIngestHistory: boolean;
+    dataQualityHistory: boolean;
+
+    // error state
+    noSelectionError: boolean;
+    mutationError: boolean;
+    showSuccessMessage: boolean;
+
+    // modal state
+    openDialog: boolean;
+};
+
+type Action =
+    | { type: 'no_selection_error' }
+    | { type: 'mutation_error' }
+    | { type: 'mutation_success' }
+    | { type: 'selection'; targetName: string; checked: boolean }
+    | { type: 'open_dialog' }
+    | { type: 'close_dialog' };
+
+const reducer = (state: State, action: Action): State => {
+    switch (action.type) {
+        case 'no_selection_error': {
+            return {
+                ...state,
+                noSelectionError: true,
+                mutationError: false,
+            };
+        }
+        case 'mutation_error': {
+            return {
+                ...state,
+                mutationError: true,
+                noSelectionError: false,
+            };
+        }
+        case 'mutation_success': {
+            return {
+                ...state,
+                // reset checkboxes
+                collectedGraphData: false,
+                dataQualityHistory: false,
+                fileIngestHistory: false,
+                highValueSelectors: false,
+
+                showSuccessMessage: true,
+            };
+        }
+        case 'selection': {
+            const { targetName, checked } = action;
+            return {
+                ...state,
+                [targetName]: checked,
+                noSelectionError: false,
+            };
+        }
+        case 'open_dialog': {
+            const { collectedGraphData, dataQualityHistory, fileIngestHistory, highValueSelectors } = state;
+            if (
+                [collectedGraphData, dataQualityHistory, fileIngestHistory, highValueSelectors].filter(Boolean)
```

Comment:
Loving the separation of concerns here in this file!

nit: Could we pull this out to a named variable to help give context as to what we are checking? Its not very obvious at first and takes a couple reads to see whats going on!

Maybe something like 
```js
const noSelection = collectedGraphData, dataQualityHistory, fileIngestHistory, highValueSelectors].filter(Boolean).length === 0

if (noSelection) {
  ...
}

```